Sammy/fix attestations

CherryKitten opened the pull request.

Commit:
fix: fix error on missing/incompatible sbom attestations

Commit:
chore: bump version to 0.2.1

jcgruenhage commented on src/sbom.rs:
```diff
@@ -79,13 +80,18 @@ async fn get_sbom(scan_target: OsString, sbom_path: PathBuf) -> Result<Value> {
+            Some(v) => v.get("SPDX"),
+            None => output.get("SPDX"),
+        }
+        .context("Image does not have compatible sbom attestation")?;
```
This will fall back to generating the sbom, right?

CherryKitten replied:
yes, if generating sboms is enabled and there is neither a cached sbom, nor an sbom attestation, it'll continue on to calling syft.

Commit:
fix: add empty default for versionInfo

We need this because running syft against a host directory with the
output format `spdx-json` adds one informational entry to the package
array, which does not contain a versionInfo field and can simply be
ignored
